File: dev/inventario_de_componentes_informaticos/report/inventario_discos/inventario_discos.py
# Copyright (c) 2024, Alejandro Hernandez and contributors
# For license information, please see license.txt

import frappe


def execute(filters=None):
	columns= [
		{
			"fieldname":"name",
			"label":"ID",
			"fieldtype":"Data",
			"width":220
		},
		{
			"fieldname":"parent",
			"label":"Parent",
			"fieldtype":"Data",
			"width":220
		},
		{
			"fieldname":"marca",
			"label":"Marca",
			"fieldtype":"Data",
			"width":220
		},
		{
			"fieldname":"modelo",
			"label":"Modelo",
			"fieldtype":"Data",
			"width":220
		},
		{
			"fieldname":"tamano",
			"label":"Tamaño",
			"fieldtype":"Data",
			"width":220
		},
		{
			"fieldname":"tipo",
			"label":"Tipo",
			"fieldtype":"Data",
			"width":220
		},
	]
	data = frappe.get_all("Discos de almacenamiento",fields=["name","parent","disco_marca.marca","disco_modelo.modelo","disco_tamano.tamano","tipo"])
	return columns, data

File: dev/inventario_de_componentes_informaticos/report/inventario_discos/inventario_discos.js
// Copyright (c) 2024, Alejandro Hernandez and contributors
// For license information, please see license.txt

frappe.query_reports["Inventario Discos"] = {
	"filters": [
		{
			fieldname: "from_date",
			label: __("De la Fecha"),
			fieldtype: "Date",
			// default: frappe.datetime.add_days(frappe.datetime.now_date(true), -40), 
            default: frappe.datetime.get_year_start,
		},
		{
			fieldname: "to_date",
			label: __("A la Fecha"),
			fieldtype: "Date",
			default: frappe.datetime.now_date(true),
		},
	]
};
